vhdl_parser/src/tokenizer.rs

```diff
@@ -401,6 +401,14 @@ pub struct Token {
     pub kind: Kind,
     pub value: Value,
     pub pos: SrcPos,
+    pub leading_comments: Vec<Comment>,
+    pub trailing_comment: Option<Comment>,
+}
+
+#[derive(PartialEq, Clone, Debug)]
+pub struct Comment {
+    pub value: Latin1String,
+    pub pos: SrcPos,
```

Comment:
The start, length fields of the SrcPos should be factored out to their own struct Position and this struct can be used here. Then we can avoid cloning the Arc.

Reply:
I made this change, but I don't understand why it was necessary.  Every Token already has a reference to source in its 'pos' field.  Why is it different if a Comment also holds a reference to source?

Reply:
Maybe it is a micro optimization but it is an extra clone. Cloning an Arc requires some kind of thread synchronization so it is something to watch out for. Maybe you can leave it as it is. I am thinking of storing a reference to the source instead in the future to avoid all clones.